tst/EnergyPlus/unit/UnitarySystem.unit.cc

```diff
@@ -10961,3 +10962,482 @@ TEST_F(EnergyPlusFixture, UnitarySystemModel_GetInputZoneEquipmentBlankCtrlZone)
+    Real64 userspecifiedFlowPerCoolingCapacityValue = thisSys->m_MaxNoCoolHeatAirVolFlow;
+    EXPECT_EQ(thisSys->m_NoCoolHeatSAFMethod, UnitarySystems::FlowPerCoolingCapacity);
+    EXPECT_EQ(userspecifiedFlowPerCoolingCapacityValue, 0.0000462180155978106);
+
+    bool FirstHVACIteration = true;
+    int AirLoopNum = 0;
+    thisSys->sizeUnitarySystem(FirstHVACIteration, AirLoopNum);
+    // check autosized cooling and heating flow rates
+    EXPECT_EQ(thisSys->m_MaxCoolAirVolFlow, 1.5);
+    EXPECT_EQ(thisSys->m_MaxHeatAirVolFlow, 1.5);
+    // check autosized no cooling and no heating flow rates
+    EXPECT_EQ(thisSys->m_MaxNoCoolHeatAirVolFlow, userspecifiedFlowPerCoolingCapacityValue * thisSys->m_DesignCoolingCapacity);
+}
+
+TEST_F( ZoneUnitarySysTest, UnitarySystemModel_getUnitarySystemInputDataTest )
+{
+
+    std::string const idf_objects = delimited_string( {
+
+        "  AirLoopHVAC:UnitarySystem,",
+        "    Unitary System Model,                   !- Name",
+        "    Load,                                   !- Control Type",
+        "    EAST ZONE,                              !- Controlling Zone or Thermostat Location",
+        "    None,                                   !- Dehumidification Control Type",
+        "    ,                                       !- Availability Schedule Name",
+        "    Node 29,                                !- Air Inlet Node Name",
+        "    Node 30,                                !- Air Outlet Node Name",
+        "    Fan:OnOff,                              !- Supply Fan Object Type",
+        "    Supply Fan,                             !- Supply Fan Name",
+        "    DrawThrough,                            !- Fan Placement",
+        "    ,                                       !- Supply Air Fan Operating Mode Schedule Name",
+        "    Coil:Heating:Water,                     !- Heating Coil Object Type",
+        "    Water Heating Coil,                     !- Heating Coil Name",
+        "    1,                                      !- DX Heating Coil Sizing Ratio",
+        "    Coil:Cooling:Water,                     !- Cooling Coil Object Type",
+        "    Water Cooling Coil,                     !- Cooling Coil Name",
+        "    No,                                     !- Use DOAS DX Cooling Coil",
+        "    7.0,                                    !- Minimum Supply Air Temperature {C}",
+        "    SensibleOnlyLoadControl,                !- Latent Load Control",
+        "    Coil:Heating:Fuel,                      !- Supplemental Heating Coil Object Type",
+        "    Supplemental Heating Coil,              !- Supplemental Heating Coil Name",
+        "    FractionOfAutosizedCoolingValue,        !- Cooling Supply Air Flow Rate Method",
+        "    ,                                       !- Cooling Supply Air Flow Rate {m3/s}",
+        "    ,                                       !- Cooling Supply Air Flow Rate Per Floor Area {m3/s-m2}",
```

Comment:
@Nigusse I was expecting to see a value in every field for this test. But now I realize that these fields are initially stored in a local variable but ultimately, one struct variable holds one of several field inputs. So now I understand why you were asking about testing every method option. At some point, we need a more robust test, but that may involve refactoring `getUnitarySystemInputData`, and you've done enough for this particular issue.

Reply:
I see the max flow rate used here and also comments as to why it's there. I guess maybe it's time to relax that constraint so that users can choose the air flow rate. I looked around and variability of air flow in cooling/heating is typical of variable speed units, [pg 4 here](https://www.shareddocs.com/hvac/docs/1009/Public/01/25VNA-05PD.pdf). I didn't see where constant speed or even 2-speed units have different flow rates in cooling/heating [pg 12 and 13 here](https://www.shareddocs.com/hvac/docs/1009/Public/0A/50VG-04SI.pdf) but I didn't spend much time looking at manufacturers data, except for choosing what I was going to say here. Maybe Carrier and Trane could add more information on choices for air flow rates.